File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        encountered = defaultdict(int)
        for num in nums:
            encountered[num] = encountered[num] + 1
        
        pq = []
        for key, value in encountered.items():
            heapq.heappush(pq, (value * -1, key))
        
        res = []
        for _ in range(k):
            value, key = heapq.heappop(pq)
            res.append(key)

        return res


        

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_map<int, bool> dupeMap;
        for (int num: nums) {
            if (dupeMap.find(num) != dupeMap.end()) {
                return true;
            }
            dupeMap[num] = true;
        }
        return false;
    }
};